test/e2e/specs/all-features-confirmation/recordset/presentation.spec.ts

```diff
@@ -0,0 +1,804 @@
+        'outbound1_outbound1 one', // all_outbound_entity_o1_o1
+        'current: outbound2_outbound1 one(12345111, 12,345,111), array: 12345221| 12345222', // all_outbound_entity_o2_o1
+        '12,345,111', // all_outbound_scalar_o1_o1
+        'current: 12,345,111, 12345111, o1_o1_o1: outbound1_outbound1_outbound1 one, o2_o1: 12,345,111', //all_outbound_scalar_o2_o1
+        'inbound1 one, inbound1 two', // array_d_entity_i1
+        'current: inbound2 one(12345221, 12,345,221), cnt: 5', // array_d_entity_i2
+        '1,234,521, 1,234,522, 1,234,523, 1,234,524, 1,234,525', // array_d_scalar_i1
+        'current: 12,345,221, 12,345,222 - 12345221| 12345222, max: 12,345,225', // array_d_scalar_i2
+        '5', //cnt_i1
+        'current: 5, 5, cnt_i1: 5, array_i3: inbound3 one', //cnt_i2
+        '5', //cnt_d_i1
+        'current: 5, 5, cnt_i1: 5, cnt_i2: 5, array_i4: i01, i02, i03, i04, i05', //cnt_d_i2
+        '1,234,521', //min_i1
+        'current: 12,345,221, 12345221, 12,345,111', //min_i2
+        '1,234,525', //max_i1
+        'current: 12,345,225, 12345225', //max_i2
+        'virtual col value is 12,345,225', //virtual column
+      ],
+      [
+        'main two', '', '1,234,502',
+        '', '', '', '', '', '', '',
+        '', '', '', '', '', '', '0',
+        '', '0', '', '', '', '', '',
+        ''
+      ]
+    ]
+  },
+  system_columns: {
+    schema_name: 'system-columns-heuristic',
+    table_name: 'system-columns',
+    compactConfig: ['RCB', 'RMT'],
+    detailedConfig: true,
+    entryConfig: ['RCB', 'RMB', 'RMT'],
+    compactColumnsSystemColumnsTable: ['id', 'text', 'int', 'RCB', 'RMT'],
+    detailedColumns: ['RID', 'id', 'text', 'int', 'RCB', 'RMB', 'RCT', 'RMT'],
+    compactColumnsPersonTable: ['id', 'text', 'RCB', 'RMT'], // no int column because it's the foreign key link (would be redundent)
+    entryColumns: ['id', 'text', 'int', 'RCB', 'RMB', 'RMT']
+  }
+};
+
+test.describe('View recordset', () => {
+  test.describe.configure({ mode: 'parallel' });
+
+  // NOTE: should this NOT run in CI still?
+  test('For recordset with columns with waitfor', async ({ page, baseURL }, testInfo) => {
+    const params = testParams.active_list;
+    const data = params.data;
+
+    const PAGE_URL = `/recordset/#${getCatalogID(testInfo.project.name)}/${params.schema_name}:${params.table_name}`;
+
+    await test.step('should load recordset page', async () => {
+      await page.goto(`${baseURL}${PAGE_URL}@sort(${params.sortby})`);
+      await RecordsetLocators.waitForRecordsetPageReady(page);
+
+      // "wait for" table column aggregate spinners to all be hidden by waiting until the count retruned in 0
+      // NOTE: using `waitForAggregates` fails because there are "multiple spinners on the page"
+      await expect.soft(RecordsetLocators.getAggregateSpinners(page)).toHaveCount(0);
```

Comment:
Oh good catch, I didn't realize `waitForAggregates` has this issue.

I don't think we should have two different ways of doing this, we should have just one `waitForAggregates` (or any other names that you see fit) that is used in all places and it should handle count. If we want to go by the way we've done this, it should most probably go to `recordset-locators`. And then maybe change the function to be `waitForRecordsetAggregates`